core/internals/internals.py

```diff
@@ -0,0 +1,46 @@
+import pkgutil
+import importlib
+import logging
+
+from mongoengine import *
+
+from core.database import YetiDocument
+from core.constants import DB_VERSION
+from core.constants import MIGRATIONS_DIRECTORY
+
+class Internals(YetiDocument):
+    db_version = IntField(default=0)
+    name = StringField(default="default", unique=True)
+    __internal = None
+
+    @classmethod
+    def syncdb(klass):
+        current_version = klass.get_internals().db_version
+        if DB_VERSION > current_version:
+            print "[+] Database version outdated: {} vs. {}".format(current_version, DB_VERSION)
+            klass.apply_migrations(current_version, DB_VERSION)
+        else:
+            print "[+] Database version is synced with code."
+
+    @classmethod
+    def get_internals(klass):
+        if klass.__internal is None:
+            klass.__internal = Internals.get_or_create(name="default")
+        return klass.__internal
+
+    @classmethod
+    def apply_migrations(klass, current_version, target_version):
+        print "    Applying migrations..."
+        print "    Current DB version: {}".format(current_version)
+        print "    Target DB version: {}".format(target_version)
```

Comment:
This might seem like a detail but I would change the terminology to remove mentions to "DB" and / or "Database". Migrations being python script, they can also take care of other tasks. Maybe talk about migration number ?